### ISSUE TITLE
Style/SafeNavigation suggested in improper context

### ISSUE THREAD

**Issue by ledbettj** [bbatsov/rubocop#3515]
The new `Style/SafeNavigation` cop is being triggered in a context that doesn't make sense, where using the safe navigation operator changes the meaning of the expression.

Given the following configuration:

```yaml
AllCops:
  TargetRubyVersion: 2.3
```

and the following file:

```ruby
require 'active_support/all'
test = [nil, '', [], {}, 'hello world']
test.delete_if { |x| x.nil? || x.empty? }
puts test
```

RuboCop suggests:

```
test.rb:3:22: C: Use safe navigation (&.) instead of checking if an object exists before calling the method.
test.delete_if { |x| x.nil? || x.empty? }
                     ^^^^^^^^^^^^^^^^^^
```
--------

## Expected behavior

It should not suggest using the safe navigation operator across a `||` expression.
`x.nil? || x.foo` is logically different than `x&.foo` when the return value is used:

```
[2] pry(main)> x&.foo
=> nil
[3] pry(main)> x.nil? || x.foo
=> true
```

## Actual behavior

RuboCop suggests using the safe navigation operator.

## Steps to reproduce the problem

Rum RuboCop 0.43.0 with the configuration provided above on the test code provided above.

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
john@nix:~/Projects/test$ rubocop -V
0.43.0 (using Parser 2.3.1.3, running on ruby 2.3.1 x86_64-darwin15)
```

**Comment by soberstadt** [comment]
How is your test code different from `x.blank?`?

**Comment by ledbettj** [comment]
It might not be, but ignore the second part of the or conditional -- it's irrelevant to the issue what that method is.

**Comment by soberstadt** [comment]
Good point, your "Expected behavior" section helps.

**Comment by rrosenblum** [comment]
This appears to be a duplicate of, or related to, #3510. The code was fixed by #3517. We no longer register an offense for any or conditions. @ledbettj, if you are able to, please test against master and let us know if you still see an issue.

**Comment by ledbettj** [comment]
Tested against 67b8df723ea1ee62d3b73ef1922a9e0964ab52f6 and the warning no longer occurs.  Thanks!
